system/Helpers/filesystem_helper.php

```diff
@@ -75,6 +75,61 @@ function directory_map(string $sourceDir, int $directoryDepth = 0, bool $hidden
 
 // ------------------------------------------------------------------------
 
+if (! function_exists('directory_mirror'))
+{
+	/**
+	 * Recursively copies the files and directories of the origin directory
+	 * into the target directory, i.e. "mirror" its contents.
+	 *
+	 * @param string $originDir
+	 * @param string $targetDir
+	 * @param bool $overwrite   Whether individual files overwrite on collision
+	 *
+	 * @return void
+	 *
+	 * @throws InvalidArgumentException
+	 */
+	function directory_mirror(string $originDir, string $targetDir, bool $overwrite = true): void
+	{
+		$originDir = rtrim($originDir, '\\/');
+		$targetDir = rtrim($targetDir, '\\/');
```

Comment:
this `$targetDir` assignment can be moved after next `if (is_dir($originDir))` as not used in `InvalidArgumentException`